init_conditions/get_subcell_agents.py

```diff
@@ -56,63 +57,117 @@ def download_max_cells_fov_images():
                 
 def sample_hex_grid(resolution = 10):
 
-    for file_path in os.listdir("aics_images/"):
+    if not Path(f"hex_seeds").is_dir():
+        os.mkdir("hex_seeds")
 
-        file_name = Path(file_path).name
-        if not "png" in file_name or "hex" in file_name:
-            continue
-        fov_image_slice = imageio.imread(f"aics_images/{file_name}")
+    # loop through all images, and sample desired zlices on hexgrid
+    for file_dir in os.listdir("aics_images/"):
 
+        # Create subdirectory for hex seed data for this image file
+        if not Path(f"hex_seeds/{file_dir}").is_dir():
+            os.mkdir(f"hex_seeds/{file_dir}")
+
+        # Get list of z values to sample
+        zmax = len(os.listdir(f"aics_images/{file_dir}")) - 2 # Subtract 2: 1 for 0-indexing and 1 for .tiff file
+        zlist = np.arange(0, zmax, resolution)
+
+        # Get list of x and y values to sample on hexgrid using a sample 2D image slice
+        sample_filename = os.listdir(f"aics_images/{file_dir}")[0]
+        fov_image_slice = imageio.imread(f"aics_images/{file_dir}/{sample_filename}")
         hex_centers, _ = create_hex_grid(
-            nx=round(fov_image_slice.shape[0] / resolution),
-            ny=round(fov_image_slice.shape[1] / resolution),
-            min_diam=resolution,
-            align_to_origin=False,
-            do_plot=False,
-        )
-        # plt.savefig(f"aics_images/hex_{file_name}",bbox_inches="tight")
+                nx=round(fov_image_slice.shape[0] / resolution),
+                ny=round(fov_image_slice.shape[1] / resolution),
+                min_diam=resolution,
+                align_to_origin=False,
+                do_plot=False,
+            )
 
+        # Initialize lists of coordinates and values to fill
         x_array = []
         y_array = []
+        z_array = []
         values = []
-        for hex_center in hex_centers[:]:
-            y_rounded = round(hex_center[1])
-            x_rounded = round(hex_center[0])
-            if y_rounded < fov_image_slice.shape[1] and x_rounded < fov_image_slice.shape[0]:
-                v = fov_image_slice[x_rounded][y_rounded]
-                if v > 0:
-                    x_array.append(x_rounded)
-                    y_array.append(y_rounded)
-                    values.append(v)
-
-        data = pd.DataFrame({"x": x_array, "y": y_array, "id": values})
-        data.to_csv(f"hex_id_data_{file_name}.csv", index=False)
-                
-        plt.scatter(x_array, y_array, c=values, cmap="jet", s=5)
-        plt.axis("equal")
-        plt.savefig(f"aics_images/hex_ids_{file_name}",bbox_inches="tight")
+
+        # Loop through files and sample 2D hexgrid on each z slice
+        for file_path in os.listdir(f"aics_images/{file_dir}"):
+
+            # Only use the png files which are the zslice images
+            if "png" in file_path:
+
+                # Get z value
```

Comment:
also, might be nice to break this chunk (lines 126-156) into a function that handles one z-slice